internal/services/batch/batch_pool_resource.go

```diff
@@ -1198,7 +1198,7 @@ func resourceBatchPoolRead(d *pluginsdk.ResourceData, meta interface{}) error {
 					d.Set("storage_image_reference", flattenBatchPoolImageReference(&config.ImageReference))
 
 					if config.LicenseType != nil {
-						d.Set("license_type", *config.LicenseType)
+						d.Set("license_type", config.LicenseType)
 					}
```

Comment:
fwiw this should always have a value set, so can be simplified to:

```suggestion
					d.Set("license_type", config.LicenseType)
```